components/Common/Select/index.tsx

```diff
@@ -0,0 +1,71 @@
+import * as Primitive from '@radix-ui/react-select';
+import { useId } from 'react';
+import { randomBytes } from 'crypto';
+import { ChevronDownIcon } from '@heroicons/react/24/outline';
+import type { FC } from 'react';
+
+import styles from './index.module.css';
+
+type SelectProps = {
+  values: string[];
+  defaultValue?: string;
+  placeholder?: string;
+  dropdownLabel?: string;
+  label?: string;
+  onChange?: (value: string) => void;
+};
+
+const Select: FC<SelectProps> = ({
+  values = [],
+  defaultValue,
+  placeholder,
+  label,
+  dropdownLabel,
+  onChange,
+}) => {
+  const id = useId();
+
+  return (
+    <div className={styles.select}>
+      {label && (
+        <label className={styles.label} htmlFor={id}>
+          {label}
+        </label>
+      )}
+      <Primitive.Root defaultValue={defaultValue} onValueChange={onChange}>
+        <Primitive.Trigger
+          className={styles.trigger}
+          aria-label={label}
+          id={id}
+        >
+          <Primitive.Value placeholder={placeholder} />
+          <ChevronDownIcon className={styles.icon} />
+        </Primitive.Trigger>
+        <Primitive.Portal>
+          <Primitive.Content className={styles.dropdown}>
+            <Primitive.Viewport>
+              <Primitive.Group>
+                {dropdownLabel && (
+                  <Primitive.Label className={`${styles.item} ${styles.label}`}>
+                    {dropdownLabel}
+                  </Primitive.Label>
+                )}
+                {values.map(value => (
+                  <Primitive.Item
+                    key={randomBytes(16).toString('hex')}
```

Comment:
Interesting way of generating indexes, but afaik, `keys` should be unique per props. So randomizing the `key` on every re-render might be good or not. On one point, it enforces that React recomputes the children's Elements, in other it might create more overhead for us.

Is this recomputing really needed?

Reply:
Thanks, I updated this part. I think only one unique identifier for each `Select` component is enough.